src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesDependabot.java

```diff
@@ -0,0 +1,148 @@
+package com.sap.sgs.phosphor.fosstars.data.github;
+
+import static com.sap.sgs.phosphor.fosstars.model.feature.oss.OssFeatures.USES_DEPENDABOT;
+
+import com.sap.sgs.phosphor.fosstars.model.Value;
+import com.sap.sgs.phosphor.fosstars.model.ValueSet;
+import java.io.IOException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Date;
+import java.util.Objects;
+import java.util.Optional;
+import org.kohsuke.github.GHCommit;
+import org.kohsuke.github.GHContent;
+import org.kohsuke.github.GHRepository;
+import org.kohsuke.github.GitHub;
+
+/**
+ * <p>This data provider checks if an open-source project on GitHub
+ * uses Dependabot, and fills out the
+ * {@link com.sap.sgs.phosphor.fosstars.model.feature.oss.OssFeatures#USES_DEPENDABOT} feature.</p>
+ *
+ * <p>First, the provider checks if a repository contains a configuration file for Dependabot.
+ * If the config exists, then the provider reports that the projects uses Dependabot.
+ * Next, the provider searches for commits from Dependabot in the commit history.
+ * If the commits are found, then the provider also reports that the project uses Dependabot.</p>
+ */
+public class UsesDependabot extends AbstractGitHubDataProvider {
+
+  /**
+   * A path to Dependabot configuration file in a repository.
+   */
+  private static final String DEPENDABOT_CONFIG = ".dependabot/config.yml";
+
+  /**
+   * A minimal number of characters in a config for Dependabot.
+   */
+  private static final int ACCEPTABLE_CONFIG_SIZE = 10;
+
+  /**
+   * A pattern to detect commits by Dependabot.
+   */
+  private static final String DEPENDABOT_PATTERN = "dependabot";
+
+  /**
+   * Period of time to be checked.
+   */
+  private static final Duration LATEST_MONTHS = Duration.ofDays(365);
+
+  /**
+   * Initializes a data provider.
+   *
+   * @param where A GitHub organization of user name.
+   * @param name A name of a repository.
+   * @param github An interface to the GitHub API.
+   */
+  public UsesDependabot(String where, String name, GitHub github) {
+    super(where, name, github);
+  }
+
+  @Override
+  public UsesDependabot update(ValueSet values) {
+    Objects.requireNonNull(values, "Hey! Values can't be null!");
+    logger.info("Checking if the project uses Dependabot ...");
+    values.update(usesDependabot());
+    return this;
+  }
+
+  /**
+   * Checks if a project has commits from Dependabot in its commit history.
+   *
+   * @return True if a commit from Dependabot was found, false otherwise.
+   */
+  private boolean hasDependabotCommits() {
+    try {
+      GHRepository repository = github.getRepository(path);
+
+      Date date = Date.from(Instant.now().minus(LATEST_MONTHS));
+      for (GHCommit commit : repository.listCommits().asList()) {
+        if (commit.getCommitDate().before(date)) {
+          break;
```

Comment:
Hmm shouldn't it be continue instead of break here?  If I understand correctly, you dont want to check for a commit if its older than few months right?

Reply:
The method takes into account only commits that were made last year. It assumes that the commits is sorted by commit dates (which is a natural order for in commit history). If it finds a commit which is older than the date, then it breaks because other commits are going to be even older.

Let me check if it's a correct assumption.

Reply:
The docs say nothing about how `listCommits()` sorts the commits. However, I found out that the GitHub API offers `since` parameter. I think it's better to use it rather than make an assumption about commits order. I'll update the code to use `GHCommitQueryBuilder`.

https://github-api.kohsuke.org/apidocs/org/kohsuke/github/GHCommitQueryBuilder.html

Reply:
Yes, this makes sense to me.

Reply:
@Sachpat I've updated the code. Please take a look.